Fix markdown h1 bug for guides

JKarlavige opened the pull request:
## What are you changing in this pull request and why?

This resolves a bug causing guide pages to crash when a markdown h1 header is added to the page. This also adds additional functionality to apply the `LifeCycle` pill to the markdown h1 on guide pages.

## Previews

Verify custom markdown h1 correctly set in upper-left section of guide page:
https://docs-getdbt-com-git-fix-lifecycle-bug-dbt-labs.vercel.app/guides/adapter-creation?step=1

Verify frontmatter title still shows correctly for other guide pages:
https://docs-getdbt-com-git-fix-lifecycle-bug-dbt-labs.vercel.app/guides/airflow-and-dbt-cloud?step=1

Verify custom markdown h1 still functions correctly for standard pages:
https://docs-getdbt-com-git-fix-lifecycle-bug-dbt-labs.vercel.app/docs/introduction

Verify frontmatter title still shows correctly for standard doc pages:
https://docs-getdbt-com-git-fix-lifecycle-bug-dbt-labs.vercel.app/docs/supported-data-platforms

## Checklist
- [x] Remove demo markdown h1 from `/adapter-creation` page
- [x] Remove demo markdown h1 from `/introduction` page

JKarlavige committed:
check if quickstartTitle const exists before adding class

JKarlavige committed:
show contentTitle in syntheticTitle if page is part of a guide

JKarlavige committed:
add demo markdown h1 to guide page

JKarlavige committed:
append lifecycle badge to markdown h1 header for guides

JKarlavige committed:
add demo custom header to standard page

JKarlavige commented on website/docs/docs/introduction.md:
Remove before merge

JKarlavige commented on website/docs/guides/adapter-creation.md:
Remove before merge

JKarlavige commented on website/src/theme/DocItem/Content/index.js:
This edit allows the markdown h1 to be applied to guide pages from the built-in `contentTitle` variable. `shouldRender` will be true if path includes `guides/`, and `contentTitle` will be populated if markdown h1 set. Otherwise the title defaults to the frontmatter title.

JKarlavige committed:
update comment

JKarlavige committed:
Merge branch 'current' into fix-lifecycle-bug

JKarlavige commented on website/src/components/quickstartTOC/index.js:
```diff
@@ -108,8 +109,24 @@ function QuickstartTOC() {
+        ".quickstart-container .step-container > h1 > span.lifecycle-badge"
+      );
+
+      if (quickstartTitle) {
```
This is where the original issue was occurring where the page would crash locally when adding a markdown h1 to a guide page. `quickstartTitle` was undefined, so when `quickstartTitle.classList.add` was called, the page would crash.

nghi-ly commented on website/docs/guides/adapter-creation.md:
```diff
@@ -11,6 +11,8 @@ level: 'Advanced'
 recently_updated: true
 ---
 
+# test h1 title <Lifecycle status="beta"/>
```
just checking out one more test

```suggestion
# test h1 title <Lifecycle status="team,enterprise"/>
```

nghi-ly committed:
Update website/docs/guides/adapter-creation.md

nghi-ly commented on website/docs/guides/adapter-creation.md:
```diff
@@ -11,6 +11,8 @@ level: 'Advanced'
 recently_updated: true
 ---
 
+# test h1 title <Lifecycle status="team,enterprise"/>
```
went ahead and tested this Lifecyle use case too

JKarlavige replied:
Nice catch @nghi-ly! I can get that updated tomorrow. I'm pulling in the HTML from the badge into the custom h1 section we're using for guides, so I should just have to update the part of the script grabbing that code to grab all badges!

nghi-ly reviewed (commented):
tysm for this pr, @JKarlavige !! 🙇‍♀️ 

i tested another Lifecycle use case with `status="team,enterprise"`. it's getting rendered with just `Team`, however, i expected it to look like this page's H1 https://docs.getdbt.com/docs/dbt-cloud-apis/overview . 

does it make sense to support this too and the other Lifecycle use cases? https://github.com/search?q=repo%3Adbt-labs%2Fdocs.getdbt.com+%3CLifecycle&type=code

mirnawong1 committed:
model query and auto-exposures beta (#5896)

this pr adds model query history and auto exposures (beta) to the doca
site for the broader beta cycle.

[docs
project](https://www.notion.so/dbtlabs/BI-auto-exposures-model-query-history-embeddable-tile-00b9411c36e748f09d16ede9136fc7ec)

outstanding questions:
- is data health tile enterprise only? or team + enterprise?
- is the beta call out (for feedback to to take part bit) correct?

[ ] PM review required
[x ] Docs review optional since beta

---------

Co-authored-by: nataliefiann <120089939+nataliefiann@users.noreply.github.com>
Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

runleonarun committed:
Update model-query-history.md (#5921)

## What are you changing in this pull request and why?

Adding one more beta we missed in [Mirna's
PR](https://github.com/dbt-labs/docs.getdbt.com/pull/5896)

## Checklist
<!--
Uncomment when publishing docs for a prerelease version of dbt:
- [ ] Add versioning components, as described in [Versioning
Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration
Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)
-->
- [ ] Review the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
so my content adheres to these guidelines.
- [ ] For [docs
versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning),
review how to [version a whole
page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version)
and [version a block of
content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this
PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links
that point to deleted pages

MiConnell committed:
Add saved_queries to example semantic_manifest file (#5922)

## What are you changing in this pull request and why?
<!---
Describe your changes and why you're making them. If related to an open 
issue or a pull request on dbt Core, then link to them here! 

To learn more about the writing conventions used in the dbt Labs docs,
see the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md).
-->
This PR adds an example saved_queries section to the semantic_manifest
file which was missing
## Checklist
<!--
Uncomment when publishing docs for a prerelease version of dbt:
- [ ] Add versioning components, as described in [Versioning
Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration
Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)
-->
- [x] Review the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
so my content adheres to these guidelines.

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

JKarlavige committed:
support multiple LifeCycle statuses for guides markdown title

JKarlavige committed:
Merge branch 'current' into fix-lifecycle-bug

JKarlavige commented:
@nghi-ly update is in place to support multiple `LifeCycle` statuses!

breezyfasano reviewed (approved):
This lgtm! Verified that the guide H1s were hidden correctly and displayed when needed. The Lifecycle elements look great too. 🚢

nghi-ly reviewed (approved):
looks awesome! thank you!

nghi-ly committed:
Merge branch 'current' into fix-lifecycle-bug

JKarlavige committed:
remove demo headers
